[Framework] Update changelog for 4.6.0 (#18351)
wazuh/wazuh: CHANGELOG.md

- Added new unit tests for GCloud integration and increased coverage to 99%. ([13573](https://github.com/wazuh/wazuh/pull/13573))
- Added new unit tests for Azure Storage integration and increased coverage to 99%. ([14104](https://github.com/wazuh/wazuh/pull/14104))
- Added new unit tests for Docker Listener integration. ([14177](https://github.com/wazuh/wazuh/pull/14177))

#### Changed

- FIM option fim_check_ignore now applies to files and directories. [#13264](https://github.com/wazuh/wazuh/pull/13264)
- Changed AWS integration to take into account user config found in the `.aws/config` file. ([#16531](https://github.com/wazuh/wazuh/pull/16531))
- Changed the calculation of timestamps in AWS and Azure modules by using UTC timezone. ([#14537](https://github.com/wazuh/wazuh/pull/14537))
- Changed the AWS integration to only show the `Skipping file with another prefix` message in debug mode. ([#15009](https://github.com/wazuh/wazuh/pull/15009))
- Changed debug level required to display CloudWatch Logs event messages. ([#14999](https://github.com/wazuh/wazuh/pull/14999))

> - Improved some error messages in the AWS integration to make them more descriptive when an exception is caught. ([#14524](https://github.com/wazuh/wazuh/pull/14524))

Reviewer: ```suggestion
- Enhanced clarity of certain error messages in the AWS integration for better exception tracing. ([#14524](https://github.com/wazuh/wazuh/pull/14524))
```

- Improved external integrations SQLite queries. ([13420](https://github.com/wazuh/wazuh/pull/13420))
- Improved items iteration for `Config` and `VPCFlow` AWS integrations. ([#16325](https://github.com/wazuh/wazuh/pull/16325))
- Unit tests have been added to the shared JSON handling library. [#14784](https://github.com/wazuh/wazuh/pull/14784)
- Unit tests have been added to the shared SQLite handling library. [#14476](https://github.com/wazuh/wazuh/pull/14476)

#### Fixed

- Fixed the architecture of the dependency URL for macOS. [#13534](https://github.com/wazuh/wazuh/pull/13534)
- Fixed a path length limitation that prevented FIM from reporting changes on Windows. [#13588](https://github.com/wazuh/wazuh/pull/13588)
- Updated the AWS integration to use the regions specified in the AWS config file when no regions are provided in `ossec.conf`. ([#14993](https://github.com/wazuh/wazuh/pull/14993))

> - Used correctly the error code `#2` for SIGINT signal in the AWS integration. ([#14850](https://github.com/wazuh/wazuh/pull/14850))

Reviewer: ```suggestion
- Corrected the error code for the SIGINT signal within the AWS integration to `#2`. ([#14850](https://github.com/wazuh/wazuh/pull/14850))
```

- Fixed the `discard_regex` functionality for the AWS GuardDuty integration. ([14740](https://github.com/wazuh/wazuh/pull/14740))
- Fixed error messages in the AWS integration when there is a `ClientError`. ([#14500](https://github.com/wazuh/wazuh/pull/14500))
- Fixed error that could lead to duplicate logs when using the same dates in the AWS integration. ([#14493](https://github.com/wazuh/wazuh/pull/14493))
- Fixed `check_bucket` method in AWS integration to be able to find logs without a folder in root. ([#16116](https://github.com/wazuh/wazuh/pull/16116))
- Added field validation for `last_date.json` in Azure Storage integration. ([#16360](https://github.com/wazuh/wazuh/pull/16360))

> - Improved exception handling when an inexistent region is given to VPCFlow AWS integration. ([#15763](https://github.com/wazuh/wazuh/pull/15763))

Reviewer: ```suggestion
- Improved handling of invalid regions given to the VPCFlow AWS integration, enhancing exception clarity. ([#15763](https://github.com/wazuh/wazuh/pull/15763))
```

- Removed duplicate methods in `AWSBucket` and reuse inherited ones from `WazuhIntegration`. ([15031](https://github.com/wazuh/wazuh/pull/15031))

### RESTful API

#### Added

- Added `POST /events` API endpoint to ingest logs through the API. ([#17670](https://github.com/wazuh/wazuh/pull/17670))
- Added `query`, `select` and `distinct` parameters to multiple endpoints. ([#17865](https://github.com/wazuh/wazuh/pull/17865))
- Added a new upgrade and migration mechanism for the RBAC database. ([#13919](https://github.com/wazuh/wazuh/pull/13919))
- Added new API configuration option to rotate log files based on a given size. ([#13654](https://github.com/wazuh/wazuh/pull/13654))
- Added `relative_dirname` parameter to GET, PUT and DELETE methods of the `/decoder/files/{filename}` and `/rule/files/{filename}` endpoints. ([#15994](https://github.com/wazuh/wazuh/issues/15994))
- Added new config option to disable uploading configurations containing the new `allow_higher_version` setting. ([#18212](https://github.com/wazuh/wazuh/pull/18212))
- Added API integration tests documentation. ([#13615](https://github.com/wazuh/wazuh/pull/13615))

#### Changed

> - Changed status code of API responses with Wazuh cluster error from 400 to 500. ([13646](https://github.com/wazuh/wazuh/pull/13646))

Reviewer: ```suggestion
- Changed the API's response status code for Wazuh cluster errors from 400 to 500. ([13646](https://github.com/wazuh/wazuh/pull/13646))
```

#### Fixed

- Fixed an unexpected behavior when using the `q` and `select` parameters in some endpoints. ([13421](https://github.com/wazuh/wazuh/pull/13421))

> - Fixed `GET /manager/configuration` API endpoint error when trying to get the vulnerability detector configuration section. ([#15203](https://github.com/wazuh/wazuh/pull/15203))

Reviewer: ```suggestion
- Resolved an issue in the GET /manager/configuration API endpoint when retrieving the vulnerability detector configuration section. ([#15203](https://github.com/wazuh/wazuh/pull/15203))
```

- Fixed `GET /agents/upgrade_result` endpoint internal error with code 1814 in large environments. ([#15152](https://github.com/wazuh/wazuh/pull/15152))

> - Improved alphanumeric_symbols regex to cover certain SCA remediation fields. ([#16756](https://github.com/wazuh/wazuh/pull/16756))

Reviewer: ```suggestion
- Enhanced the alphanumeric_symbols regex to better accommodate specific SCA remediation fields. ([#16756](https://github.com/wazuh/wazuh/pull/16756))
```

- Fixed bug that would not allow retrieving the Wazuh logs if only the JSON format was configured. ([#15967](https://github.com/wazuh/wazuh/pull/15967))
- Fixed error in `GET /rules` when variables are used inside `id` or `level` ruleset fields. ([#16310](https://github.com/wazuh/wazuh/pull/16310))
- Fixed `PUT /syscheck` and `PUT /rootcheck` endpoints to exclude exception codes properly. ([#16248](https://github.com/wazuh/wazuh/pull/16248))

> - Modified `test_agent_PUT_endpoints.tavern.yaml` to fix a race condition error. ([#16347](https://github.com/wazuh/wazuh/issues/16347))

Reviewer: ```suggestion
- Adjusted test_agent_PUT_endpoints.tavern.yaml to resolve a race condition error.  ([#16347](https://github.com/wazuh/wazuh/issues/16347))
```
